page-tracking/src/page_tracker.rs

```diff
@@ -193,6 +204,22 @@ impl PageTracker {
         Ok(unsafe { P::MappablePage::new_with_size(page.addr(), page.size()) })
     }
 
+    /// Consumes the shared `page`, returning a page that can then be mapped into
+    /// a page table.
+    pub fn share_page_for_mapping<P>(&self, page: P, owner: PageOwnerId) -> Result<P::MappablePage>
```

Comment:
nit: just `share_page()`. the `for_mapping` is redundant since all we can do with shared pages is map them.

Reply:
Changed.